app/helpers/hosts_and_hostgroups_helper.rb

```diff
@@ -8,7 +8,19 @@ def model_name(host)
   end
 
   def accessible_hostgroups
-    Hostgroup.with_taxonomy_scope_override(@location,@organization).order(:title)
+    Hostgroup.with_taxonomy_scope_override(@location,@organization).authorized(:view_hostgroups).reorder(:title)
+  end
+
+  def accessible_compute_resources
```

Comment:
Looks like using a form object and storing these helpers there would be easier to understand